onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -919,23 +947,41 @@ Status QnnBackendManager::ExtractProfilingEventBasic(
   std::string message = GetEventTypeString(event_data.type);
   std::string unit = GetUnitString(event_data.unit);
 
-  outfile << "UNKNOWN"
-          << ","
-          << message << ","
-          << event_data.value << ","
-          << unit << ","
-          << "BACKEND"
-          << ","
-          << eventLevel << ","
-          << (event_data.identifier ? event_data.identifier : "NULL") << "\n";
+#ifndef _WIN32
+  tracelogging_provider_ep_enabled = false;
+#endif
+
+  if (!tracelogging_provider_ep_enabled) {
+    outfile << "UNKNOWN"
+            << ","
+            << message << ","
+            << event_data.value << ","
+            << unit << ","
+            << "BACKEND"
+            << ","
+            << eventLevel << ","
+            << (event_data.identifier ? event_data.identifier : "NULL") << "\n";
+  } else {
+#ifdef _WIN32
+    LogQnnProfileEventAsTraceLogging(
+        (uint64_t)0,
+        message,
+        std::to_string(event_data.value),
+        unit,
+        "BACKEND",
+        eventLevel,
+        (event_data.identifier ? event_data.identifier : "NULL"));
+#endif
+  }
```

Comment:
seems to me, these equal to:
#ifdef _WIN32
    LogQnnProfileEventAsTraceLogging(
        (uint64_t)0,
        message,
        std::to_string(event_data.value),
        unit,
        "BACKEND",
        eventLevel,
        (event_data.identifier ? event_data.identifier : "NULL"));
#else
    outfile << "UNKNOWN"
            << ","
            << message << ","
            << event_data.value << ","
            << unit << ","
            << "BACKEND"
            << ","
            << eventLevel << ","
            << (event_data.identifier ? event_data.identifier : "NULL") << "\n";
#endif

Reply:
@HectorSVC did you account for the case where the dev specified profiling but ETW was not dynamically listening? Then we want to continue to log out the events in CSV format. The behavior would be documented once the PR is complete.

Reply:
I see

---
In reply to: [1447868177](https://github.com/microsoft/onnxruntime/pull/18882#discussion_r1447868177) [](ancestors = 1447868177)